sql/flow.go

```diff
@@ -0,0 +1,129 @@
+package sql
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/stdcopy"
+)
+
+func getPypiVersion() string {
+	url := "https://pypi.org/pypi/astro-sql-cli/json"
+	method := "GET"
+
+	httpClient := &http.Client{}
+	req, err := http.NewRequest(method, url, http.NoBody)
+	if err != nil {
+		fmt.Println(err)
+		return ""
+	}
+	res, err := httpClient.Do(req)
+	if err != nil {
+		fmt.Println(err)
+		return ""
+	}
+	defer res.Body.Close()
+
+	type response struct {
+		Info struct {
+			Version string `json:"version"`
+		} `json:"info"`
+	}
+	var resp response
+	err = json.NewDecoder(res.Body).Decode(&resp)
+	if err != nil {
+		fmt.Println(err)
+		return ""
+	}
+
+	return resp.Info.Version
+}
+
+func getContext(filePath string) io.Reader {
+	ctx, _ := archive.TarWithOptions(filePath, &archive.TarOptions{})
+	return ctx
+}
+
+func CommonDockerUtil(cmd []string, flags map[string]string) error {
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		panic(err)
+	}
+
+	astroSQLCliVersion := getPypiVersion()
+	opts := types.ImageBuildOptions{
+		Dockerfile: "Dockerfile.sql_cli",
+		Tags:       []string{"sql_cli"},
+	}
+
+	if astroSQLCliVersion != "" {
+		opts.BuildArgs = map[string]*string{"VERSION": &astroSQLCliVersion}
+	}
+
+	_, currentfilePath, _, ok := runtime.Caller(0)
+	if !ok {
+		fmt.Println("Failed to get current file path and hence cannot locate Dockerfile for building the image")
+	}
+	dockerfilePath := path.Join(path.Dir(currentfilePath), "../Dockerfile.sql_cli")
```

Comment:
I feel it is extremely wrong to assume that Docerfile will always have a fixed relative location from the CLI binary. Can we at least set an environment variable whose value is the location of the Dockerfile, and then read the value of that env variable over here?

Reply:
make sense, made the needed change.

Reply:
Can we not replicate what we do with `astro dev init` which creates a `Dockerfile`.

Only difference being we can put it in a specific folder/path.

Asking users to set an environment variable, makes the experience poor and we don't users to know about the Dockerfile at all.